apps/console/src/features/connections/components/wizards/authenticator-create-wizard.tsx

```diff
@@ -212,22 +218,33 @@ export const AuthenticatorCreateWizard: FunctionComponent<AddAuthenticatorWizard
      * @param identityProvider - Identity provider data.
      */
     const handleWizardFormFinish = (identityProvider: ConnectionInterface): void => {
-        const authenticator = identityProvider?.federatedAuthenticators?.authenticators.find((a) =>
-            a.authenticatorId === identityProvider?.federatedAuthenticators?.defaultAuthenticatorId);
+        const authenticator: FederatedAuthenticatorListItemInterface =
+            identityProvider?.federatedAuthenticators?.authenticators.find(
+                (a: FederatedAuthenticatorListItemInterface) =>
+                    a.authenticatorId === identityProvider?.federatedAuthenticators?.defaultAuthenticatorId
+            );
+
+        authenticator.properties = authenticator?.properties.filter(
+            (property: CommonPluggableComponentPropertyInterface) => {
+                if (isEmpty(property.key)) {
+                    return false;
+                }
 
-        authenticator.properties = authenticator.properties.filter((property) => {
-            if (isEmpty(property.key)) {
-                return false;
+                return true;
             }
-
-            return true;
-        });
+        );
         authenticator.properties.filter(Boolean);
-        authenticator.isDefault = false;
+
+        /**
+         * If the default authenticator is not available then the currently adding
+         * new authenticator is set as the default authenticator.
+         */
+        authenticator.isDefault = !isDefaultAuthenticatorAvailable;
```

Comment:
Wouldn't it be more clearer to directly assign `true` to isDefault?

Reply:
It can also be `false`. Therefore we need to keep this logic.